src/butil/iobuf_inl.cpp

```diff
@@ -0,0 +1,116 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
1. The file name `iobuf_inl` means there are inline functions in it. It is always for header file, not cpp file. You can just move these code to `iobuf_inl.h`, and delete this `iobuf_inl.cpp` file. 
2. When you seperate the function body from class/struct definition, that function is by default not inline. This may affect performance. To make sure the function is inline, you can keep the function body in class/struct definition, or add `inline` keyword for each function.